inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -18,79 +18,288 @@
+            
+            logger.debug(f"Added repeated schedule with id: {str(schedule_id)}, request_id:{rs.request_id}, cron_duration: {rs.cron_duration}, cron_minutes: {rs.cron_minutes}, cron_hours: {rs.cron_hours}, cron_day_month: {rs.cron_day_month}, cron_month: {rs.cron_month}, cron_day_week: {rs.cron_day_week}") # noqa
 
-        # Add the task
-        for manifest in task.manifests:
-            sql = ''' INSERT INTO single_task(request_id, start_time, end_time, manifests)
-                                  VALUES(?,?,?,?) '''
-            logger.debug(f"Adding task: {str(task.request_id), str(task.start_time), str(task.end_time), str(manifest)}")
-            cur.execute(sql, (str(task.request_id), str(task.start_time), str(task.end_time), str(manifest)))
-            conn.commit()
-        logger.debug("Task added.")
+            # Add the manifests to the manifest table
+            manifest_ids = self._insert_manifest_to_table(rs.manifests)
+            
+            # Add the schedule_id and manifest_id to the repeated_schedule_manifest table
+            self._insert_repeated_schedule_manifest_tables(schedule_id, manifest_ids)
+            self._conn.commit()
+        except (sqlite3.Error) as e:
+            self._conn.rollback()
+            logger.error(f"Transaction failed: {str(e)}")
+            raise DispatcherException(f"Transaction failed: {str(e)}")        
+        
+    def _insert_manifest_to_table(self, manifests: list[str]) -> list[int]:
+        # Add the manifest to the manifest table
+        if len(manifests) == 0:
+            raise DispatcherException("Error: At least one manifest is required for the schedule.  Manifests list is empty.")
 
-    def _create_repeated_schedule_task(self, task: RepeatedSchedule, conn: sqlite3.Connection) -> None:
-        # Creating the table
-        cur = conn.cursor()
-        # Check if the table already exists, if not, create the table
-        logger.debug("Checking if the table exist...")
-        cur.execute("SELECT name FROM sqlite_master WHERE type='table' AND name='repeated_task'")
-        if cur.fetchone() is None:
-            logger.debug("Table not exist. Creating the table.")
-            sql = ''' CREATE TABLE repeated_task (
-                                    request_id text,
-                                    cron_duration text,
-                                    cron_minutes text,
-                                    cron_hours text,
-                                    cron_day_month text,
-                                    cron_month text,
-                                    cron_day_week text,
-                                    manifests text
-                                ); '''
-            conn.execute(sql)  # Execute the SQL command
+        manifest_ids: list[int] = []        
+                
+        for manifest in manifests:
+            sql = ''' INSERT INTO manifest(manifest) VALUES(?); '''
```

Comment:
should this be wrapped in a transaction so we don't get some of the inserts succeeding and some failing?

Reply:
I believe it is.  The call to this method is wrapped in a transaction.  In _create_repeated_schedule and _create_single_schedule.